### ISSUE TITLE
Add support for the Motorola moto g45 5G

### ISSUE THREAD

**Issue by quickfixdemon** [PrintHelloPeople/android_device_motorola_fogos-twrp#4]
Hello,

I've opened this issue to ask if support for the Motorola Moto g45 5G can be added. According to AIDA64, the Motorola Moto g45 5G reports the Board and Device parameters as fogos, Product as fogos_gp, and the Platform as holi. Essentially, the Moto g45 5G seems to be a repackaged Moto g34, as they share similar a chipset and specifications. So, if it is possible, could you add TWRP support for the Moto g45? I'd be willing to test it on my device if needed.

Thank you.

**Comment by PrintHelloPeople** [comment]
HI,
Have you tried booting (fastboot boot imagename.img) the TWRP image of Motorola G34 5G in your device? Tell me if it boots and if present any issues 

Thanks

**Comment by quickfixdemon** [comment]
> HI, Have you tried booting (fastboot boot imagename.img) the TWRP image of Motorola G34 5G in your device? Tell me if it boots and if present any issues
> 
> Thanks

I haven't tried it yet, I'll see if I can unlock the bootloader and flash the TWRP image this week. However, I'd just like to ask, in case anything goes wrong and my phone bricks itself, is there any way I can recover it?

**Comment by PrintHelloPeople** [comment]
When you boot an image it doesn't change any partition in your phone, worst case scenario is a reboot to bootloader.
I want you to know that there's already a group in Telegram for G34 and G45 where different ROM are released.
Here's the link: https://t.me/motog34updates

**Comment by PrintHelloPeople** [comment]
The TWRP listed in that group is the TWRP present in this repo.

**Comment by quickfixdemon** [comment]
> When you boot an image it doesn't change any partition in your phone, worst case scenario is a reboot to bootloader. I want you to know that there's already a group in Telegram for G34 and G45 where different ROM are released. Here's the link: https://t.me/motog34updates

OK, but according to Motorola, they'll stop providing software updates after I unlock the bootloader. Would it be a good idea to try installing the updates regardless of that? I would really love to install a custom ROM on my device, but I've heard that certain apps and things might not work, e.g., banking or financial apps(plus the LineageOS support for the g34/45 is unofficial). I wouldn't have any of these issues or hesitation if the g45 was EOL, but it is still due to receive the android 15 update. So, would this still be advisable to do? Or should I wait until android 15 releases for the g45?

I'm really sorry to annoy you with all these questions, its just that I'm concerned that I might lose my only working phone.

**Comment by PrintHelloPeople** [comment]
Don't worry, ask as many question as you want!
After you unlock the bootloader you erase the warranty, in short if you break your phone it's only your fault. It's not true that you won't receive any update after that. I used to experiment different ROMs for the past six months, after that I re installed the stock ROM and re-locked the bootloader, now I'm receiving again Motorola updates (indeed I'm running latest Android security patches). When you are running custom ROM you won't receive any Motorola updates because it's not official software. 
If G45 is your main phone and you heavily depend on financial apps (bank apps, Google wallet and others) I don't recommend you experimenting with custom ROMs. Yes banking apps are problematic in custom ROMs due to play integrity checks. Some ROMs like lineage OS are shipped as signed, that means that you SHOULD pass play integrity by default (not always working because a lot of different factors can disrupt it). If you are planning to root the custom ROM it will make the use of banking apps even more difficult, you'll need to use modules like PlayIntegrityFix to pass play integrity but they depend on Google will, that in anytime can do a mass ban and make those modules stop working and you'll need to wait for those modules to update.
But custom ROMs have not only disadvantages but also some advantages like: better performances, better battery and software.
In short if you don't have any major issue, like bad battery and performance or big bugs, just keep the bootloader locked and wait for Android 15 you'll save yourself moments of frustration (personal experience).

**Comment by quickfixdemon** [comment]
@PrintHelloPeople hello again! I just wanted to ask another question. I was discussing porting pmOS to my smartphone yesterday and one of the people I was talking to mentioned something about an EDL loader for Qualcomm devices. Apparently if you can get the EDL loader for a device it becomes unbrickable(according to said person). He gave a link to a github repo containing EDL loaders for a lot of devices, including Motorola smartphones. I went through it and found a loader for Motorola snapdragon 695 devices. However, the person said the EDL loader is chip specific and device specific, so I'm not sure if the loader I found will work on my smartphone. I wanted to ask you if you could check whether the EDL loader I found would work on a moto g34 and possibly the moto g45. Here is the git repo(https://github.com/bkerler/Loaders/tree/a83850577f32faaf4694cda8f00666e0d9d34654/lenovo_motorola) and the name of the loader i think might work for my device(01a90e102e80000_c0122e0f6ceb5bad_fhprg.bin) I'm not sure if this is even the correct loader or not, I've chosen this one because the last change on it said something like Add support for Motorola 695 based devices.

**Comment by Anonymous941** [comment]
The EDL has to be unlocked, or have a known exploit.  Most of them are locked down and require a confidential private key.  Companies can't change this without just giving you the key, because it's stored in ROM.

That being said, if you don't use `fastboot flashing unlock_critical` and only `fastboot flashing unlock`, you'll _probably_ be able to recover from bricks as it won't let you overwrite the bootloader

**Comment by quickfixdemon** [comment]
> Don't worry, ask as many question as you want! After you unlock the bootloader you erase the warranty, in short if you break your phone it's only your fault. It's not true that you won't receive any update after that. I used to experiment different ROMs for the past six months, after that I re installed the stock ROM and re-locked the bootloader, now I'm receiving again Motorola updates (indeed I'm running latest Android security patches). When you are running custom ROM you won't receive any Motorola updates because it's not official software. If G45 is your main phone and you heavily depend on financial apps (bank apps, Google wallet and others) I don't recommend you experimenting with custom ROMs. Yes banking apps are problematic in custom ROMs due to play integrity checks. Some ROMs like lineage OS are shipped as signed, that means that you SHOULD pass play integrity by default (not always working because a lot of different factors can disrupt it). If you are planning to root the custom ROM it will make the use of banking apps even more difficult, you'll need to use modules like PlayIntegrityFix to pass play integrity but they depend on Google will, that in anytime can do a mass ban and make those modules stop working and you'll need to wait for those modules to update. But custom ROMs have not only disadvantages but also some advantages like: better performances, better battery and software. In short if you don't have any major issue, like bad battery and performance or big bugs, just keep the bootloader locked and wait for Android 15 you'll save yourself moments of frustration (personal experience).

Hello again! Long time no see.

My phone recently gained official LineageOS support so I've decided to flash it and use it as a daily driver. Screw my warranty, it can go to hell for all I care. I remembered this issue and thought I'd come back to this. I've decided to test the TWRP build, so what should I look out for when booting TWRP? What should I test for you to make this an official TWRP build? Please let me know.
